src/mute/__tests__/muteModel-test.js

```diff
@@ -2,37 +2,106 @@
-import { getMute, reducer } from '../muteModel';
-import { EVENT_MUTED_TOPICS } from '../../actionConstants';
+import { getMute, getTopicVisibilityPolicy, isTopicMuted, reducer } from '../muteModel';
+import { EVENT, EVENT_MUTED_TOPICS } from '../../actionConstants';
 import * as eg from '../../__tests__/lib/exampleData';
-import { makeMuteState } from './mute-testlib';
+import { makeMuteState, makeUserTopic } from './mute-testlib';
 import { tryGetActiveAccountState } from '../../selectors';
+import { UserTopicVisibilityPolicy } from '../../api/modelTypes';
+import { EventTypes } from '../../api/eventTypes';
+
+/* eslint jest/expect-expect: ["error", { "assertFunctionNames": ["expect", "check"] }] */
 
 const initialState = makeMuteState([]);
 
+describe('getters', () => {
+  describe('getTopicVisibilityPolicy', () => {
+    function check(state, expected) {
+      expect(getTopicVisibilityPolicy(state, eg.stream.stream_id, 'topic')).toEqual(expected);
+    }
+
+    test('with nothing for stream', () => {
+      check(makeMuteState([]), UserTopicVisibilityPolicy.None);
+    });
+
+    test('with nothing for topic', () => {
+      check(makeMuteState([[eg.stream, 'other topic']]), UserTopicVisibilityPolicy.None);
+    });
+
+    test('with topic muted', () => {
+      check(makeMuteState([[eg.stream, 'topic']]), UserTopicVisibilityPolicy.Muted);
+    });
+  });
+
+  describe('isTopicMuted', () => {
+    function check(state, expected) {
+      expect(isTopicMuted(eg.stream.stream_id, 'topic', state)).toEqual(expected);
+    }
+
+    test('with nothing for stream', () => {
+      check(makeMuteState([]), false);
+    });
+
+    test('with nothing for topic', () => {
+      check(makeMuteState([[eg.stream, 'other topic']]), false);
+    });
+
+    test('with topic muted', () => {
+      check(makeMuteState([[eg.stream, 'topic']]), true);
+    });
+  });
+});
+
 describe('reducer', () => {
   describe('REGISTER_COMPLETE', () => {
-    test('when mute data is provided init state with it: local', () => {
-      const action = eg.mkActionRegisterComplete({ muted_topics: [[eg.stream.name, 'topic']] });
+    test('in modern user_topics format: unit test', () => {
+      const action = eg.mkActionRegisterComplete({
+        user_topics: [makeUserTopic(eg.stream, 'topic', UserTopicVisibilityPolicy.Muted)],
+      });
       expect(reducer(initialState, action, eg.plusReduxState)).toEqual(
         makeMuteState([[eg.stream, 'topic']]),
       );
     });
 
-    test('when mute data is provided init state with it: end-to-end', () => {
+    test('in modern user_topics format: end-to-end test', () => {
       const action = eg.mkActionRegisterComplete({
         streams: [eg.stream],
         subscriptions: [eg.subscription],
+        user_topics: [makeUserTopic(eg.stream, 'topic', UserTopicVisibilityPolicy.Muted)],
+      });
+      const newState = tryGetActiveAccountState(fullReducer(eg.baseReduxState, action));
+      expect(newState).toBeTruthy();
```

Comment:
Can the mute-model code (what this file is responsible for testing) cause `newState` to be falsy? I wonder if this could just be an `invariant`, as here: https://github.com/zulip/zulip-mobile/blob/41e50a740/src/__tests__/permissionSelectors-test.js#L41

Here, and in another case below.

Reply:
It could. I don't totally love this `toBeTruthy` solution, because it doesn't narrow the type and so the next line still has to account for the falsy case:
```js
      expect(newState).toBeTruthy();
      expect(newState && getMute(newState)).toEqual(makeMuteState([[eg.stream, 'topic']]));
```

But I don't love the `invariant` solution either, here in a test. The Jest `expect` API is nice because if the assertion is ever wrong, it gives good informative error output that's helpful for debugging. Also `invariant` requires a message argument, which is a good thing for error cases in application code -- helpful for sorting out what's happening from a Sentry event or whatever, and for clarifying why the author thought it was an invariant -- but feels redundant here.

I'd probably go for `invariant` if the narrowing is more necessary, like if the subsequent logic that relied on the condition was more complex so that repeatedly accounting for it was more of a pain. If so, I might go for something like `invariant(newState, '');`, leaving the message empty.